core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -2408,7 +2408,10 @@ private void applyLagToLastPartition(long maxTimestamp, int lagRowCount, long la
         }
         // set append position on columns so that the files are truncated to the correct size
         // if the partition is closed after the commit.
-        setAppendPosition(txWriter.transientRowCount, false);
+        // If wal commit terminates here, column positions should include lag to not truncate the WAL lag data.
+        // Otherwise, lag will be copied out and ok to truncate to the transient row count.
+        long partitionTruncateRowCount = txWriter.getTransientRowCount() + (commitTerminates ? txWriter.getLagRowCount() : 0);
+        setAppendPosition(partitionTruncateRowCount, false);
```

Comment:
`WalTableSqlTest#testRenameDropTable()` seem to hit the `commitTerminates && txWriter.getLagRowCount() > 0` condition reliably.